src/serialization/SerializationService.ts

```diff
@@ -291,44 +325,36 @@ export class SerializationServiceV1 implements SerializationService {
+        this.registerSerializer(String.prototype.constructor, new StringSerializer(), new StringArraySerializer());
+        this.registerSerializer(Number.prototype.constructor, new DoubleSerializer(), new DoubleArraySerializer());
+        this.registerSerializer(SerializationSymbols.BYTE_SYMBOL , new ByteSerializer(), new ByteArraySerializer());
+        this.registerSerializer(Boolean.prototype.constructor, new BooleanSerializer(), new BooleanArraySerializer());
+        this.registerSerializer(SerializationSymbols.NULL_SYMBOL, new NullSerializer(), null);
+        this.registerSerializer(SerializationSymbols.SHORT_SYMBOL, new ShortSerializer(), new ShortArraySerializer());
+        this.registerSerializer(SerializationSymbols.INTEGER_SYMBOL, new IntegerSerializer(), new IntegerArraySerializer());
+        this.registerSerializer(Long.prototype.constructor, new LongSerializer(), new LongArraySerializer());
+        this.registerSerializer(SerializationSymbols.FLOAT_SYMBOL, new FloatSerializer(), new FloatArraySerializer());
+        this.registerSerializer(SerializationSymbols.CHAR_SYMBOL, new CharSerializer(), new CharArraySerializer());
+        this.registerSerializer(Date.prototype.constructor, new DateSerializer(), null);
+        this.registerSerializer(LocalDate.prototype.constructor, new LocalDateSerializer(), null);
+        this.registerSerializer(LocalTime.prototype.constructor, new LocalTimeSerializer(), null);
+        this.registerSerializer(LocalDateTime.prototype.constructor, new LocalDateTimeSerializer(), null);
+        this.registerSerializer(OffsetDateTime.prototype.constructor, new OffsetDateTimeSerializer(), null);
+        this.registerSerializer(SerializationSymbols.JAVACLASS_SYMBOL, new JavaClassSerializer(), null);
+        this.registerSerializer(SerializationSymbols.ARRAYLIST_SYMBOL, new ArrayListSerializer(), null);
+        this.registerSerializer(SerializationSymbols.LINKLIST_SYMBOL, new LinkedListSerializer(), null);
+        this.registerSerializer(UUID.prototype.constructor, new UuidSerializer(), null);
+        this.registerSerializer(BigDecimal.prototype.constructor, new BigDecimalSerializer(), null);
+        this.registerSerializer(BigInt.prototype.constructor, new BigIntSerializer(), null);
+        this.registerSerializer(SerializationSymbols.JAVA_ARRAY_SYMBOL, new JavaArraySerializer(), null);
+        this.registerSerializer(SerializationSymbols.COMPACT_SYMBOL, this.compactStreamSerializer, null);
+        this.registerSerializer(SerializationSymbols.IDENTIFIED_SYMBOL, this.identifiedSerializer, null);
+        this.registerSerializer(SerializationSymbols.PORTABLE_SYMBOL, this.portableSerializer, null);
+
         if (this.serializationConfig.jsonStringDeserializationPolicy === JsonStringDeserializationPolicy.EAGER) {
-            this.registerSerializer('!json', new JsonSerializer());
+            this.registerSerializer(SerializationSymbols.JSON_SYMBOL, new JsonSerializer(), null);
```

Comment:
Since the parameter is optional, why not pass nothing? Also I think we should stick to `undefined` or null for the values that do not exist. you can make sure we don't use null. If you don't pass anything to a function that has optional parameter it will be undefined. 

Similar for all registerSerializer calls, we can remove third parameters